packages/common/src/lib/core/utils/utils.ts

```diff
@@ -34,19 +34,19 @@ export function getHttpProxyAgent(proxyUrl?: string): HttpProxyAgent | HttpsProx
     console.log(`The proxy server at '${proxy}' will be used.`);
   }
   
-  if (proxy?.startsWith('https')) {
+  if (downloadUrl?.startsWith('https')) {
     return new HttpsProxyAgent(proxyAgentOpts);
-  } else if(proxy?.startsWith('http')){
+  } else if(downloadUrl?.startsWith('http')){
```

Comment:
change the condition to:
```
  if (targetUrl.startsWith('http')  && !!proxyUrl) {
```

(_only set the proxy if needed (proxyUrl  is truthy)_)